File: src/main/java/gg/mineral/practice/commands/spectator/StopSpectatingCommand.java
package gg.mineral.practice.commands.spectator;

import gg.mineral.practice.commands.PlayerCommand;

import gg.mineral.practice.managers.ProfileManager;
import lombok.val;

public class StopSpectatingCommand extends PlayerCommand {

    public StopSpectatingCommand() {
        super("stopspectating");
        setAliases("spec");
    }

    @Override
    public void execute(org.bukkit.entity.Player pl, String[] args) {
        val profile = ProfileManager.getOrCreateProfile(pl);
        profile.getSpectateHandler().stopSpectating();
    }
}

File: src/main/kotlin/gg/mineral/practice/commands/spectator/StopSpectatingCommand.kt
package gg.mineral.practice.commands.spectator

import dev.rollczi.litecommands.annotations.command.Command
import dev.rollczi.litecommands.annotations.context.Context
import dev.rollczi.litecommands.annotations.execute.Execute
import gg.mineral.practice.entity.Profile

@Command(name = "stopspectating", aliases = ["stopspec"])
class StopSpectatingCommand {
    @Execute
    fun execute(@Context profile: Profile) = profile.stopSpectating()
}
